docker/docker.go

```diff
@@ -136,3 +133,13 @@ func getSwarmConfiguration(config *agent.RuntimeConfiguration, dockerInfo types.
 
 	return nil
 }
+
+func getCli() (*client.Client, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithVersion(agent.SupportedDockerAPIVersion))
+	if err != nil {
+		return nil, err
+	}
+	defer cli.Close()
```

Comment:
This `defer` does not belong here, as it will close the client right after returning it.